ocaml/typing/typetexp.ml

```diff
@@ -541,6 +541,12 @@ let check_arg_type styp =
     | _ -> ()
   end
 
+let layout_of_annotation_or_attr annot_ctx attrs layout =
```

Comment:
The `_or_attr` here confused me: it's saying that the layout comes from the attribute, but that's not quite accurate. I would just go with `layout_of_annotation`.